src/main/java/com/direwolf20/buildinggadgets/common/schema/template/TranslationWrapper.java

```diff
@@ -0,0 +1,42 @@
+package com.direwolf20.buildinggadgets.common.schema.template;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockPos.Mutable;
+import net.minecraft.util.math.Vec3i;
+
+import java.util.Iterator;
+
+public final class TranslationWrapper implements Iterable<TemplateData> {
```

Comment:
This class is not used? What the intention of it?

Reply:
Wait and see....

You need to be able to translate somewhere in order to build at the right spot.